docs/design/2020-09-30-index-usage-information.md

```diff
@@ -0,0 +1,166 @@
+# Proposal:
+
+- Author(s):     [rebelice](https://github.com/rebelice)
+- Last updated:  Sep. 30, 2020
+- Discussion at: N/A
+
+## Abstract
+
+This document describes the design of recording index usage information.
+
+## Background
+
+There may be unused indexes in the database. In addition, modifying database tables, indexes, and query statements may cause some indexes to not be used in the future. Users may want to clear these long-unused indexes to relieve storage and performance pressure.
+
+Related issues:
+
+- https://github.com/pingcap/tidb/issues/14998
+- https://github.com/pingcap/tidb/issues/14607
+- https://github.com/pingcap/tidb/issues/17508
+- https://asktug.com/t/topic/34614/12
+
+## Proposal
+
+### SCHEMA_INDEX_USAGE
+
+Design system tables to record index usage information. The system table is designed as follows:
+
+1. The system table is located in `mysql` database and is named `SCHEMA_INDEX_USAGE`.
+2. Columns of `SCHEMA_INDEX_USAGE`:
+
+| Column name  | Data type | Description |
+|--------------|-----------|-------------|
+| TABLE_SCHEMA | varchar   | Name of the database on which the table or view is defined.|
+| TABLE_NAME   | varchar   | Name of the table or view on which the index is defined.|
+| INDEX_NAME   | varchar   | Name of the index.|
+| QUERY_COUNT  | longlong  | Number of the SQL using this index.|
+| ROWS_SELECTED| longlong  | Number of rows read from the index. We can check the average fetched rows count of each query of the index through `ROWS_READ` / `QUERY_COUNT`.|
+| LAST_USED_AT | timestamp | The last time of the SQL using this index.|
+
+
+3. Typical usage: `select * from mysql.SCHEMA_INDEX_USAGE`.
+
+#### Table creation: 
+
+```sql
+create table SCHEMA_INDEX_USAGE (
+	TABLE_SCHEMA varchar(64),
+	TABLE_NAME varchar(64),
```

Comment:
We'd better store table id and index id?
The table name and index name can be changed by ddl stmt.

Reply:
I think name is more user-friendly than id. And there are two solutions to this problem. One is to handle the DDL of rename to modify the name in the table, and the other is to create a more user-friendly view based on this table. I prefer the latter. If you think so, I'll modify it in next PR and write it in the document.

Reply:
The second one LGTM.